internal/cmd/log.go

```diff
@@ -38,7 +39,16 @@ func logStatus(env Env, msg string, a ...interface{}) {
 	if !ok {
 		format = defaultLogFormat
 	}
-	if format != "" {
+	shouldLog := true
+	filter, ok := env["DIRENV_LOG_FILTER"]
+	if ok {
+		filterRegEx, err := regexp.Compile(filter)
```

Comment:
If you want to use it on both sides, then this has to use the same regexp engine as bash. I believe it's PCRE.

Alternatively, it would be acceptable to introduce a `direnv log` command to invoke on the stdlib side.